docs/src/pages/components/stack/stack.md

```diff
@@ -0,0 +1,39 @@
+---
+title: React Stack component
+components: Stack
+githubLabel: 'component: Stack'
+---
+
+# Stack
+
+<p class="description">The Stack component manages children along the vertical and horizontal axes with optional spacing between children.</p>
+
+{{"component": "modules/components/ComponentLinkHeader.js"}}
+
+## Usage
+
+`Stack` is concerned with one-dimensional layouts compared to [`Grid`](/components/grid/) that handles two-dimensional layouts. The default direction is `column` which stacks children vertically. To control space between children, use the `spacing` prop.
+
+{{"demo": "pages/components/stack/IntroStack.js", "bg": true}}
+
+## Direction
+
+By default, the `Stack` positions items in a `column`. You can change the direction to a `row`:
+
+{{"demo": "pages/components/stack/DirectionStack.js", "bg": true}}
+
+## Dividers
+
+Use the `divider` prop to insert an element between each child.
+
+{{"demo": "pages/components/stack/DividerStack.js", "bg": true}}
+
+## Responsive values
+
+Easily switch out `direction` or `spacing` based on the current active breakpoint. Note, that each breakpoint is required to define both properties.
```

Comment:
This is a drawback mentioned [here](https://github.com/mui-org/material-ui/pull/25149#issuecomment-791276880). Right now we can't read the current set of styles easily at a specific breakpoint since we need to use the current `direction` when calculating the correct side to apply margin. I think this could be fixed in `handleBreakpoints` in a future PR, but will take some reworking. Happy to expand on this in the description. Possible we could even type that for now?

Reply:
Taking the example from the docs:

```jsx
<Stack
  direction={{ xs: 'column', md: 'row', lg: 'row' }}
  spacing={{ xs: 1, md: 2, lg: 4 }}
>
```
If `lg` is omitted from direction, would it not be `row` by default? (`i.e. md` is `md` and up?) Similarly, if `lg` is omitted for spacing instead, could it not use the `md` spacing?

Reply:
I can think of 3 possible solutions here.

1. We document the limitation and add a warning when each breakpoint isn't provided. It's wouldn't lead to an awesome DX, but do we need to care this much?
2. We add an ad-hoc logic to resolve, for each spacing breakpoint, what's the direction
3. We build the underlying function needed for #23885. Basically, take the current breakpoint in, and return what's the correct value. We use this in the 

```diff
diff --git a/packages/material-ui/src/Stack/Stack.js b/packages/material-ui/src/Stack/Stack.js
index a18ecb3065..8979a5c55a 100644
--- a/packages/material-ui/src/Stack/Stack.js
+++ b/packages/material-ui/src/Stack/Stack.js
@@ -31,6 +31,20 @@ function intersperse(children, separator) {
   }, []);
 }

+function resolveBreakpointValues({ values, theme }) {
+  let previous;
+
+  return Object.keys(theme.breakpoints.values).reduce((acc, breakpoint) => {
+    if (typeof values === 'string') {
+      acc[breakpoint] = values;
+    } else {
+      acc[breakpoint] = values[breakpoint] || values[previous];
+    }
+    previous = breakpoint;
+    return acc;
+  }, {});
+}
+
 const getSideFromDirection = (direction) => {
   switch (direction) {
     case 'row':
@@ -55,12 +69,16 @@ export const style = ({ styleProps, theme }) => {

   if (styleProps.spacing) {
     const transformer = createUnarySpacing(theme);
+    const directionValues = resolveBreakpointValues({ values: styleProps.direction, theme });
+
     const styleFromPropValue = (propValue, breakpoint) => {
-      const direction = styleProps.direction[breakpoint] || styleProps.direction;
       return {
         '& > :not(styles) + :not(styles)': {
           margin: 0,
-          [`margin${getSideFromDirection(direction)}`]: getValue(transformer, propValue),
+          [`margin${getSideFromDirection(directionValues[breakpoint])}`]: getValue(
+            transformer,
+            propValue,
+          ),
         },
       };
     };
diff --git a/packages/material-ui/src/Stack/Stack.test.js b/packages/material-ui/src/Stack/Stack.test.js
index e954c6515b..148d743f82 100644
--- a/packages/material-ui/src/Stack/Stack.test.js
+++ b/packages/material-ui/src/Stack/Stack.test.js
@@ -24,7 +24,7 @@ describe('<Stack />', () => {
     expect(
       style({
         styleProps: {
-          direction: { xs: 'column', sm: 'row', md: 'row' },
+          direction: { xs: 'column', sm: 'row' },
           spacing: { xs: 1, sm: 2, md: 4 },
         },
         theme,
@@ -49,7 +49,6 @@ describe('<Stack />', () => {
           margin: 0,
           marginLeft: '32px',
         },
-        flexDirection: 'row',
       },
       display: 'flex',
     });
```

Reply:
I would vote for option 3 😁

Reply:
Oh no, my bad option 3 is wrong.

Reply:
Option 1 might be simpler 🤔

Reply:
Ouch, it took me 1 hour, but it seems to be fixed now.